server/support_packet.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"path/filepath"
+	"sort"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/plugin"
+)
+
+func (p *Plugin) GenerateSupportData(_ *plugin.Context) ([]*model.FileData, error) {
+	jobs, err := p.GetAllJobs(context.TODO())
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve jobs")
+	}
+
+	jobSlice := make([]*DumpJob, 0, len(jobs))
+	for _, job := range jobs {
+		jobSlice = append(jobSlice, job)
+	}
+
+	sort.Slice(jobSlice, func(i, j int) bool {
+		return jobSlice[i].CreateAt > jobSlice[j].CreateAt
+	})
+
+	var recentJob *DumpJob
+	for _, j := range jobSlice {
+		if j.Status == model.JobStatusSuccess {
+			recentJob = j
+			break
+		}
+	}
+
+	if recentJob == nil {
+		return nil, errors.New("there is no dumps in the filestore, please create a tsdb dump first")
+	}
+	dumpLocation := recentJob.DumpLocation
+
+	fr, err := p.fileBackend.Reader(dumpLocation)
+	if err != nil {
+		return nil, fmt.Errorf("could not read dump: %w", err)
+	}
+	defer fr.Close()
+
+	b, err := io.ReadAll(fr)
```

Comment:
Are we sure this won't cause OOM issues? Ideally we should stream the file contents downstream right?

Reply:
Yes, there is a small possibility but this was how the API being implemented, we need to change the API to accept something like a `io.ReaderCloser`. However or in our case; ideally server can directly read from the filestore but still the API needs to be changed.

Reply:
Ok, wondering if we should file a ticket to track this? For example, k8s pods have a mem limit to them. There could be cases where this might be problematic. I'll approve for now but we should probably have a streaming API.